api/server/httputils/errors.go

```diff
@@ -129,3 +133,24 @@ func statusCodeFromGRPCError(err error) int {
 		return http.StatusInternalServerError
 	}
 }
+
+// statusCodeFromDistributionError returns status code according to registry errcode
+// code is loosely based on errcode.ServeJSON() in docker/distribution
+func statusCodeFromDistributionError(err error) int {
```

Comment:
Perhaps we should have a utility in github.com/docker/distribution to work with these errors; they're a bit hard to use (imo)